addon/model/controller.ts

```diff
@@ -0,0 +1,69 @@
+import Command from "@lblod/ember-rdfa-editor/commands/command";
+import {AnyEventName, EditorEventListener, ListenerConfig} from "@lblod/ember-rdfa-editor/utils/event-bus";
+import ModelSelection from "@lblod/ember-rdfa-editor/model/model-selection";
+import RawEditor from "@lblod/ember-rdfa-editor/utils/ce/raw-editor";
+import {NotImplementedError} from "@lblod/ember-rdfa-editor/utils/errors";
+
+export type WidgetLocation = "toolbar" | "sidebar";
+
+export interface WidgetSpec {
+  componentName: string;
+  desiredLocation: WidgetLocation;
+}
+
+export default interface Controller {
+  get name(): string;
+
+  get selection(): ModelSelection;
+
+  executeCommand<A extends unknown[], R>(commandName: string, ...args: A): R | void;
+
+  registerCommand<A extends unknown[], R>(command: Command<A, R>): void;
+
+  registerWidget(spec: WidgetSpec): void;
+
+  onEvent<E extends AnyEventName>(eventName: E, callback: EditorEventListener<E>, config?: ListenerConfig): void;
+
+  offEvent<E extends AnyEventName>(eventName: E, callback: EditorEventListener<E>, config?: ListenerConfig): void;
+
+}
+
+export class RawEditorController implements Controller {
+  private readonly _name: string;
+  private readonly _rawEditor: RawEditor;
+
+  constructor(name: string, rawEditor: RawEditor) {
+    this._name = name;
+    this._rawEditor = rawEditor;
+  }
+
+  get name(): string {
+    return this._name;
+  }
+
+  get selection(): ModelSelection {
+    return this._rawEditor.selection;
+  }
+
+  executeCommand<A extends unknown[], R>(commandName: string, ...args: A): R | void {
+    return this._rawEditor.executeCommand(commandName, ...args);
+  }
+
+  offEvent<E extends AnyEventName>(eventName: E, callback: EditorEventListener<E>, config?: ListenerConfig): void {
```

Comment:
What's the meaning of this function? Not sure I understand

Reply:
to remove an event listener